### ISSUE TITLE
ohai WARN: unable to detect ipaddress, macaddress

### ISSUE THREAD

**Issue by ghost** [chef/ohai#397]
Ohai: 7.2.4
app chef # ohai > test.ohai
[2014-08-23T05:08:39+03:00] WARN: unable to detect ipaddress
[2014-08-23T05:08:39+03:00] WARN: unable to detect macaddress
app chef # 

it did at least detect the gateway:

```
"network": {
    "interfaces": {
      "eth0:": {
        "type": "eth",
        "number": "0:"
      },
      "lo:": {

      },
      "sit0:": {
        "type": "sit",
        "number": "0:"
      }
    },
    "default_gateway": "172.16.20.1",
    "default_interface": "eth0"
  },
  "counters": {
    "network": {
      "interfaces": {

      }
    }
  },
```

network info : 
```
app chef # ip link 
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT 
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP mode DEFAULT qlen 1000
    link/ether 52:54:00:70:26:0b brd ff:ff:ff:ff:ff:ff
3: sit0: <NOARP> mtu 1480 qdisc noop state DOWN mode DEFAULT 
    link/sit 0.0.0.0 brd 0.0.0.0

app chef # ip addr      
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN 
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 52:54:00:70:26:0b brd ff:ff:ff:ff:ff:ff
    inet 172.16.20.2/24 brd 172.16.20.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fe80::5054:ff:fe70:260b/64 scope link 
       valid_lft forever preferred_lft forever
3: sit0: <NOARP> mtu 1480 qdisc noop state DOWN 
    link/sit 0.0.0.0 brd 0.0.0.0
app chef # 

app chef # ifconfig
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.16.20.2  netmask 255.255.255.0  broadcast 172.16.20.255
        inet6 fe80::5054:ff:fe70:260b  prefixlen 64  scopeid 0x20<link>
        ether 52:54:00:70:26:0b  txqueuelen 1000  (Ethernet)
        RX packets 368449  bytes 169286648 (161.4 MiB)
        RX errors 0  dropped 201227  overruns 0  frame 0
        TX packets 132059  bytes 68529387 (65.3 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 0  (Local Loopback)
        RX packets 2556  bytes 249426 (243.5 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 2556  bytes 249426 (243.5 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

app chef # 
```

other info 
```
ruby 1.9.3p547 (2014-05-14 revision 45962) [x86_64-linux]
app chef # gem --version
2.2.2
app chef # gem list --local

*** LOCAL GEMS ***

bigdecimal (1.1.0)
bundler (1.7.2)
bundler-unload (1.0.2)
chef (11.14.6)
chef-zero (2.2)
coderay (1.1.0)
diff-lcs (1.2.5)
erubis (2.7.0)
executable-hooks (1.3.2)
ffi (1.9.3)
ffi-yajl (1.0.2)
gem-wrappers (1.2.5)
hashie (2.1.2)
highline (1.6.21)
io-console (0.3)
ipaddress (0.8.0)
json (1.5.5)
libyajl2 (1.0.1)
method_source (0.8.2)
mime-types (1.25.1)
minitest (2.5.1)
mixlib-authentication (1.3.0)
mixlib-cli (1.5.0)
mixlib-config (2.1.0)
mixlib-log (1.6.0)
mixlib-shellout (1.6.0.rc.0)
net-ssh (2.9.1)
net-ssh-gateway (1.2.0)
net-ssh-multi (1.2.0)
ohai (7.2.4)
plist (3.1.0)
pry (0.10.1)
rack (1.6.0.beta)
rake (0.9.2.2)
rdoc (3.9.5)
rest-client (1.6.7)
rubygems-bundler (1.4.4)
rvm (1.11.3.9)
slop (3.6.0)
systemu (2.6.4)
wmi-lite (1.0.0)
app chef # 
```

**Comment by jmara** [comment]
Slightly different but ipaddress/macaddress and the whole network Mash is not detected on XenServer 6.5 which is Centos 5.10 based. Interface names have not changed, but "ip addr" acts weird :-/ 
```
# ip -V
ip utility, iproute2-ss121001
```

Any ideas?

**Comment by lamont-granquist** [comment]
XenServer 6.5 is probably #343 

Please try to keep issues on one track.

**Comment by jmara** [comment]
@lamont-granquist Sorry you are totally right, I only searched for Xen and #343 didn't pop up. I'll be more carefully in the future :)

**Comment by tas50** [comment]
I'd love if someone with a Gentoo box would try out https://github.com/chef/ohai/pull/510 and let me know if that fixes your issue.  The fix is based on the path that @BugRoger gave.

**Comment by cmluciano** [comment]
A fix for gentoo/coreos is merged in #510. Can this issue be closed?
